metacat-main/src/main/java/com/netflix/metacat/main/services/impl/TableServiceImpl.java

```diff
@@ -437,7 +551,24 @@ public void rename(
         if (oldTable != null) {
             //Ignore if the operation is not supported, so that we can at least go ahead and save the user metadata
             eventBus.post(new MetacatRenameTablePreEvent(oldName, metacatRequestContext, this, newName));
-            connectorTableServiceProxy.rename(oldName, newName, isMView);
+
+            // Before rename, first rename its parent child relation
+            parentChildRelMetadataService.rename(oldName, newName);
+
+            try {
+                connectorTableServiceProxy.rename(oldName, newName, isMView);
+            } catch (Exception e) {
+                try {
+                    // if rename operation fail, rename back the parent child relation
+                    parentChildRelMetadataService.rename(newName, oldName);
```

Comment:
If making the two stage commit changes, in first step insert a relation with the new name, then in this step if exception thrown delete the relation with the new name.

If no exception then delete the relation with the old name.